mitmproxy/tools/web/app.py

```diff
@@ -453,6 +453,15 @@ def get(self, flow_id, message, content_view):
         ))
 
 
+class Commands(RequestHandler):
+    def post(self):
+        result = self.master.commands.execute(self.json["command"])
+        if result is None:
+            self.write({"result": ""})
+            return
+        self.write({"result": str(result)})
```

Comment:
Is there a reason why we `str` the result? It would be nice to eventually support more datatypes here, so we want to generally aim for arbitrary JSON.

Reply:
If we don't `str` the result, the state in CommandBar component doesn't store the result. I don't know the reason of this behavior.

Reply:
This works fine for me for everything that is json-serializable, e.g. `view.order.options` executes fine. It fails for flows, e.g. `view.flows.resolve`:
```
Traceback (most recent call last):                                                                                      
  File "c:\python39\lib\site-packages\tornado\web.py", line 1702, in _execute                                           
    result = method(*self.path_args, **self.path_kwargs)                                                                
  File "c:\users\user\git\mitmproxy\mitmproxy\tools\web\app.py", line 462, in post                                      
    self.write({"result": result})                                                                                      
  File "c:\users\user\git\mitmproxy\mitmproxy\tools\web\app.py", line 158, in write                                     
    super().write(chunk)                                                                                                
  File "c:\python39\lib\site-packages\tornado\web.py", line 843, in write                                               
    chunk = escape.json_encode(chunk)                                                                                   
  File "c:\python39\lib\site-packages\tornado\escape.py", line 75, in json_encode                                       
    return json.dumps(value).replace("</", "<\\/")                                                                      
  File "c:\python39\lib\json\__init__.py", line 231, in dumps    return _default_encoder.encode(obj)                    
  File "c:\python39\lib\json\encoder.py", line 199, in encode    chunks = self.iterencode(o, _one_shot=True)            
  File "c:\python39\lib\json\encoder.py", line 257, in iterencode                                                       
    return _iterencode(o, 0)                                                                                            
  File "c:\python39\lib\json\encoder.py", line 179, in default                                                          
    raise TypeError(f'Object of type {o.__class__.__name__} 'TypeError: Object of type HTTPFlow is not JSON serializable
```

Long story short, I think it's best if we return a `{value, type}` object here. For example:
```python
# a string is returned, for example from `view.order`
{
  value: "foo"
  type: "str",
}
# nothing is returned, for example from `view.clear`
{
  value: null
  type: "None",
}
# a list of flows is returned, for example from `view.flows.resolve @all`
{
  value: [flowId1, flowId2, ...],
  type: "list[flow]",
}
# a list of strings is returned, for example from flow.encode.options
{
  value: ["foo", "bar"],
  type: "list[str]",
}
```

Does that make sense?

Reply:
Sorry, it was a bug in my CommandBar component.
It didn't display the result when command returned 0.
Now, I fixed it and applied your suggested change.